### PR TITLE
add set facility rewrite op

### PR DESCRIPTION
**Pull request by bazsi**
This PR is a followup to #3115 namely that added set-severity() and this one adds set-facility().
Apart from the new feature, this also refactors the set-severity() test program and adds a log message to set-severity(), similar to what we have in other rewrite ops (and similar to what I have added to set-facility() as  a part of this PR).

Hopefully the gods of travis et al find these patches worthwhile and give me the holy price of being "green". Apart from that this should be pretty straight-forward to get merged.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by furiel** [comment]
Could you resolve the conflicts?

Could you summarize what work is left to be done, or is there any open questions?

**Comment by bazsi** [comment]
rebased & added unit test.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: lib/syslog-names.h [hunk 1]
```diff
@@ -36,6 +36,9 @@
 /* extract priority */
 #define LOG_PRI(p)  ((p) & LOG_PRIMASK)
 
+#define SEVERITY_CODE(n)    ((n) & 7)
```
Comment (Kokan) [inline lib/syslog-names.h]: These are not really needed to be visible, could we keep them local to the original compilation unit ?

File: lib/rewrite/rewrite-set-facility.c [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  LogRewriteSetFacility *self = (LogRewriteSetFacility *) s;
+
+  log_msg_make_writable(pmsg, path_options);
+
+  log_template_format(self->facility, *pmsg, NULL, LTZ_LOCAL, 0, NULL, result);
+
+  const gint facility = _convert_facility(result->str);
+  if (facility < 0)
+    {
+      msg_debug("Warning: invalid value passed to set-facility()",
```
Comment (furiel) [inline lib/rewrite/rewrite-set-facility.c:96]: This should be an error message.

File: lib/rewrite/rewrite-set-facility.c [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+static LogPipe *
+log_rewrite_set_facility_clone(LogPipe *s)
+{
+  LogRewriteSetFacility *self = (LogRewriteSetFacility *) s;
+  LogRewriteSetFacility *cloned = (LogRewriteSetFacility *)log_rewrite_set_facility_new(log_template_ref(self->facility),
+                                  s->cfg);
+
+  if (self->super.condition)
+    cloned->super.condition = filter_expr_ref(self->super.condition);
```
Comment (furiel) [inline lib/rewrite/rewrite-set-facility.c:120]: `rewrite_condition_opt` is missing from grammar.

File: lib/rewrite/rewrite-set-facility.c (hunk 2 above)
Comment (bazsi) [reply lib/rewrite/rewrite-set-facility.c:96]: This is the same as in set-severity () where I asked to make this debug.

This is a per message path and I would rather ignore this as an error than generate shitloads of errors in case a parser goes wrong or it an attacker generates incorrect value on purpose.

We should not trust the value too much.

File: lib/rewrite/rewrite-set-facility.c (hunk 3 above)
Comment (bazsi) [reply lib/rewrite/rewrite-set-facility.c:120]: Thanks for spotting this. Probably it is missing in set-severity () too, as I was borrowing code from that.

**Comment by bazsi** [comment]
We discussed that I would do the conditional rewrite part work.

**Comment by bazsi** [comment]
rebased, and added condition() support for set-facility()/set-severity() as well as clear-tag() (where it was also missing).

**Comment by bazsi** [comment]
setting milestone, as I think this is ready to go in.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by furiel** [comment]
@bazsi travis fails on one commit: a signed off is missing. Other than that, I checked the patchet, and it looks good to me.

**Comment by bazsi** [comment]
done. hopefully this time it's going to be green.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS
